param/parameterized.py

```diff
@@ -2481,9 +2481,12 @@ def __init__(self, **params):
                 # instantiation of Parameterized with watched deps. Will
                 # probably store expanded deps on class - see metaclass
                 # 'dependers'.
-                for p in self.param.params_depended_on(n):
+                grouped = defaultdict(list)
+                for dep in self.param.params_depended_on(n):
+                    grouped[(id(dep.inst),id(dep.cls),dep.what)].append(dep)
+                for group in grouped.values():
                     # TODO: can't remember why not just pass m (rather than _m_caller) here
-                    (p.inst or p.cls).param.watch(_m_caller(self, n), p.name, p.what, queued=queued)
+                    (dep.inst or dep.cls).param.watch(_m_caller(self, n), [dep.name for dep in group], dep.what, queued=queued)
```

Comment:
I'm confused by this line: `dep` (in `dep.inst, dep.cls` and `dep.what`) looks like they refer to the leaky loop variable left over from the previous `for dep in self.param.params_depended_on(n)` loop.

Reply:
```suggestion
                    gdep = group[0] # Need to grab representative dep from this group
                    (gdep.inst or gdep.cls).param.watch(_m_caller(self, n), [d.name for d in group], gdep.what, queued=queued)
```

Reply:
🤦 ha ha

My excuse is all I ever saw was core dumps :)